Formula/postgresql@11.rb

```diff
@@ -0,0 +1,106 @@
+class PostgresqlAT11 < Formula
+  desc "Object-relational database system"
+  homepage "https://www.postgresql.org/"
+  url "https://ftp.postgresql.org/pub/source/v11.5/postgresql-11.5.tar.bz2"
+  sha256 "7fdf23060bfc715144cbf2696cf05b0fa284ad3eb21f0c378591c6bca99ad180"
+  head "https://github.com/postgres/postgres.git"
+
+  keg_only :versioned_formula
+
+  depends_on "pkg-config" => :build
+  depends_on "icu4c"
+  depends_on "openssl@1.1"
+  depends_on "readline"
+
+  def install
+    # avoid adding the SDK library directory to the linker search path
+    ENV["XML2_CONFIG"] = "xml2-config --exec-prefix=/usr"
+
+    ENV.prepend "LDFLAGS", "-L#{Formula["openssl@1.1"].opt_lib} -L#{Formula["readline"].opt_lib}"
+    ENV.prepend "CPPFLAGS", "-I#{Formula["openssl@1.1"].opt_include} -I#{Formula["readline"].opt_include}"
+
+    args = %W[
+      --disable-debug
+      --prefix=#{prefix}
+      --datadir=#{HOMEBREW_PREFIX}/share/postgresql
+      --libdir=#{HOMEBREW_PREFIX}/lib
+      --sysconfdir=#{etc}
+      --docdir=#{doc}
+      --enable-thread-safety
+      --with-bonjour
+      --with-gssapi
+      --with-icu
+      --with-ldap
+      --with-libxml
+      --with-libxslt
+      --with-openssl
+      --with-pam
+      --with-perl
+      --with-uuid=e2fs
+    ]
+
+    # The CLT is required to build Tcl support on 10.7 and 10.8 because
+    # tclConfig.sh is not part of the SDK
+    args << "--with-tcl"
+    if File.exist?("#{MacOS.sdk_path}/System/Library/Frameworks/Tcl.framework/tclConfig.sh")
+      args << "--with-tclconfig=#{MacOS.sdk_path}/System/Library/Frameworks/Tcl.framework"
+    end
+
+    system "./configure", *args
+    system "make"
+    system "make", "install-world", "datadir=#{pkgshare}",
+                                    "libdir=#{lib}",
+                                    "pkglibdir=#{lib}/postgresql"
+  end
+
+  def post_install
+    (var/"log").mkpath
+    (var/"postgres").mkpath
+    unless File.exist? "#{var}/postgres/PG_VERSION"
+      system "#{bin}/initdb", "--locale=C", "-E", "UTF-8", "#{var}/postgres"
+    end
+  end
+
+  def caveats; <<~EOS
+    To migrate existing data from a previous major version of PostgreSQL run:
+      brew postgresql-upgrade-database
+  EOS
+  end
+
+  plist_options :manual => "pg_ctl -D #{HOMEBREW_PREFIX}/var/postgres start"
```

Comment:
This also needs to be updated for the versioned formula. If you `diff` the `postgres` formula and the `postgres@10` there might be more clues as to what changed between the last time we bumped the major version of postgres.

Reply:
I'll continue to work through the issues.

I'm now seeing the commands that are running in CI that show failing tests, it looks like for at least some of them I can reproduce locally and shorten the feedback loop.

When I did the audit and build from source (when originally opening the PR) it was all for postgres 12 - I didn't run any real tests for postgresql@11 (at that time)